src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Renamer.cs

```diff
@@ -107,14 +110,12 @@ public void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs args)
                                 return;
                             }
                         }
-                    }
+                    }                    
 
                     // Now do the rename
                     var optionSet = newDocument.Project.Solution.Workspace.Options;
-                    var renamedSolution = await RoslyRenamer.Renamer.RenameSymbolAsync(newDocument.Project.Solution, symbol, newName, optionSet).ConfigureAwait(false);
-
-                    await _threadingService.SwitchToUIThread();
-
+                    var renamedSolution = await RoslynRenamer.Renamer.RenameSymbolAsync(newDocument.Project.Solution, symbol, newName, optionSet).ConfigureAwait(true);
```

Comment:
Can you add a comment here that TryApplyChanges needs to occur on the UI thread? It's not obvious, for example, I could move this RoslynRenamer call to a different location, not knowing that I've just changed behavior to the next call.